src/components/container/RelativeActivityWeekNavigator/RelativeActivityWeekNavigator.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import { DailyDataQueryResult, DailyDataType, queryDailyData } from "../../../helpers/query-daily-data";
+import { ColorDefinition } from "../../../helpers/colors";
+import { SparkBarChart, SparkBarChartBar, WeekCalendar } from "../../presentational";
+import { add, addDays, startOfDay } from "date-fns";
+import { useInitializeView } from "../../../helpers/Initialization";
+import getDayKey from "../../../helpers/get-day-key";
+
+export interface RelativeActivityWeekNavigatorProps {
+    selectedDate: Date;
+    onDateSelected(date: Date): void;
+    dataTypes: WeeklyRelativeActivityDataType[];
+    previewState?: "default";
+    innerRef?: React.Ref<HTMLDivElement>;
+}
+
+export interface WeeklyRelativeActivityDataType {
+    dailyDataType: string;
+    threshold: number;
+    color?: ColorDefinition;
+    overThresholdColor?: ColorDefinition;
+}
+
+export default function (props: RelativeActivityWeekNavigatorProps) {
+    let [weekStart, setWeekStart] = useState<Date>(add(startOfDay(new Date()), { days: -6 }));
+    let [dailyData, setDailyData] = useState<{ [key: string]: DailyDataQueryResult } | null>(null);
+
+    function loadData() {
+        if (!props.dataTypes.length) return;
+
+        function queryData() {
+            if (props.previewState === "default") {
+                let result = props.dataTypes.map(dataType => {
+                    let data: DailyDataQueryResult = {};
+                    for (let i = -7; i < 7; i++) {
+                        let dayKey = getDayKey(add(weekStart, { days: i }));
+                        data[dayKey] = Math.random() * dataType.threshold * 2;
+                    }
+                    return data;
+                });
+                return Promise.resolve(result);
+            }
+
+            let promises = props.dataTypes.map(dataType => queryDailyData(dataType.dailyDataType, add(weekStart, { days: -7 }), add(weekStart, { days: 7 })));
+            return Promise.all(promises).then(results => {
+                return results;
+            });
+        }
+
+        queryData().then((results) => {
+            let resultsMap: { [key: string]: DailyDataQueryResult } = { ...dailyData };
+            let currentDailyData = dailyData || {};
+            results.forEach((result, index) => {
+                let newResult = { ...currentDailyData[props.dataTypes[index].dailyDataType], ...result };
+                resultsMap[props.dataTypes[index].dailyDataType] = newResult;
+            });
+            setDailyData(resultsMap);
+        });
+
+        if (props.previewState === "default") {
```

Comment:
Is this code block necessary?  Seems like the daily data is already going to get set appropriately when in the preview state.

Reply:
Removed as part of refactoring